lib/core.js

```diff
@@ -1,8 +1,58 @@
 'use strict';
 
 const ApiGroup = require('./api-group');
+const fs = require('fs');
+
+const certPath = '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt';
+const tokenPath = '/var/run/secrets/kubernetes.io/serviceaccount/token';
+const namespacePath = '/var/run/secrets/kubernetes.io/serviceaccount/namespace';
 
 class Core extends ApiGroup {
+  /**
+  * Returns with in cluster config
+  * Based on: https://github.com/kubernetes/client-go/blob/124670e99da15091e13916f0ad4b2b2df2a39cd5/rest/config.go#L274
+  * and http://kubernetes.io/docs/user-guide/accessing-the-cluster/#accessing-the-api-from-a-pod
+  *
+  * @static getInCluster
+  * @returns {Object} { url, cert, auth, namespace }
+  */
+  static getInClusterConfig () {
+    const host = process.env.KUBERNETES_SERVICE_HOST;
+    const port = process.env.KUBERNETES_SERVICE_PORT;
+    let cert;
+    let namespace;
+    let bearer;
+
+    if (!host || !port) {
+      throw TypeError('Unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined');
+    }
+
+    try {
```

Comment:
I'd simplify this by not handling each `readFileSync` exception individually because I think there's enough context in those `Error`s:
```
const cert = fs.readFileSync(certPath, 'utf8');
const bearer = fs.readFileSync(tokenPath, 'utf8');
const namespace = fs.readFileSync(namespacePath, 'utf8');
```